teamcity: failed tests on master: acceptance/TestVersionUpgrade

The following tests appear to have failed:

[#583670](https://teamcity.cockroachdb.com/viewLog.html?buildId=583670):


```
--- FAIL: acceptance/TestVersionUpgrade (67.450s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade109470162
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestVersionUpgrade: TestVersionUpgrade/runMode=local (67.440s)
version_upgrade_test.go:270: binary=v1.1.0
version_upgrade_test.go:270: cluster=1.0
version_upgrade_test.go:270: cluster=1.0-3
version_upgrade_test.go:270: binary=v1.1.1
version_upgrade_test.go:270: cluster=1.1
version_upgrade_test.go:270: binary=source
version_upgrade_test.go:270: cluster=1.1-6
version_upgrade_test.go:191: 1: pq: gossip and KV store disagree about value
------- Stdout: -------
CockroachDB node starting at 2018-03-26 15:28:32.268990225 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:36297
sql:        postgresql://root@127.0.0.1:37045?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:     initialized new cluster
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     1
CockroachDB node starting at 2018-03-26 15:28:32.663205108 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:39527
sql:        postgresql://root@127.0.0.1:33703?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:     initialized new node, joined pre-existing cluster
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     2
CockroachDB node starting at 2018-03-26 15:28:32.66551453 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:45493
sql:        postgresql://root@127.0.0.1:39655?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:     initialized new node, joined pre-existing cluster
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     3
CockroachDB node starting at 2018-03-26 15:28:41.107744793 +0000 UTC (took 6.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:40565
sql:        postgresql://root@127.0.0.1:34835?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     1
CockroachDB node starting at 2018-03-26 15:28:42.879265285 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:41029
sql:        postgresql://root@127.0.0.1:41531?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     3
CockroachDB node starting at 2018-03-26 15:28:44.844552603 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:37179
sql:        postgresql://root@127.0.0.1:45287?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     2
CockroachDB node starting at 2018-03-26 15:28:52.259176919 +0000 UTC (took 3.3s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:35415
sql:        postgresql://root@127.0.0.1:45477?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     1
CockroachDB node starting at 2018-03-26 15:28:54.123698069 +0000 UTC (took 0.6s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:37777
sql:        postgresql://root@127.0.0.1:43223?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     3
CockroachDB node starting at 2018-03-26 15:29:11.039707485 +0000 UTC (took 15.5s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:40447
sql:        postgresql://root@127.0.0.1:33683?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     2
CockroachDB node starting at 2018-03-26 15:29:22.547685106 +0000 UTC (took 4.7s)
build:               CCL v2.0-alpha.20180212-1888-g4b166b7 @ 2018/03/26 15:15:43 (go1.10)
admin:               http://127.0.0.1:46169
sql:                 postgresql://root@127.0.0.1:37841?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1/cockroach-temp942803217
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:              restarted pre-existing node
clusterID:           7eab080d-d52d-4336-baae-332a937d264f
nodeID:              1
CockroachDB node starting at 2018-03-26 15:29:24.299732675 +0000 UTC (took 0.7s)
build:               CCL v2.0-alpha.20180212-1888-g4b166b7 @ 2018/03/26 15:15:43 (go1.10)
admin:               http://127.0.0.1:45079
sql:                 postgresql://root@127.0.0.1:35899?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2/cockroach-temp516241572
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:              restarted pre-existing node
clusterID:           7eab080d-d52d-4336-baae-332a937d264f
nodeID:              3
CockroachDB node starting at 2018-03-26 15:29:26.334185802 +0000 UTC (took 0.6s)
build:               CCL v2.0-alpha.20180212-1888-g4b166b7 @ 2018/03/26 15:15:43 (go1.10)
admin:               http://127.0.0.1:41959
sql:                 postgresql://root@127.0.0.1:33935?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3/cockroach-temp821448691
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:              restarted pre-existing node
clusterID:           7eab080d-d52d-4336-baae-332a937d264f
nodeID:              2
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade109470162
--- FAIL: acceptance/TestVersionUpgrade (67.450s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade109470162
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestVersionUpgrade: TestVersionUpgrade/runMode=local (67.440s)
version_upgrade_test.go:270: binary=v1.1.0
version_upgrade_test.go:270: cluster=1.0
version_upgrade_test.go:270: cluster=1.0-3
version_upgrade_test.go:270: binary=v1.1.1
version_upgrade_test.go:270: cluster=1.1
version_upgrade_test.go:270: binary=source
version_upgrade_test.go:270: cluster=1.1-6
version_upgrade_test.go:191: 1: pq: gossip and KV store disagree about value
------- Stdout: -------
CockroachDB node starting at 2018-03-26 15:28:32.268990225 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:36297
sql:        postgresql://root@127.0.0.1:37045?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:     initialized new cluster
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     1
CockroachDB node starting at 2018-03-26 15:28:32.663205108 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:39527
sql:        postgresql://root@127.0.0.1:33703?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:     initialized new node, joined pre-existing cluster
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     2
CockroachDB node starting at 2018-03-26 15:28:32.66551453 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:45493
sql:        postgresql://root@127.0.0.1:39655?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:     initialized new node, joined pre-existing cluster
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     3
CockroachDB node starting at 2018-03-26 15:28:41.107744793 +0000 UTC (took 6.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:40565
sql:        postgresql://root@127.0.0.1:34835?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     1
CockroachDB node starting at 2018-03-26 15:28:42.879265285 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:41029
sql:        postgresql://root@127.0.0.1:41531?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     3
CockroachDB node starting at 2018-03-26 15:28:44.844552603 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:37179
sql:        postgresql://root@127.0.0.1:45287?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     2
CockroachDB node starting at 2018-03-26 15:28:52.259176919 +0000 UTC (took 3.3s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:35415
sql:        postgresql://root@127.0.0.1:45477?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     1
CockroachDB node starting at 2018-03-26 15:28:54.123698069 +0000 UTC (took 0.6s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:37777
sql:        postgresql://root@127.0.0.1:43223?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     3
CockroachDB node starting at 2018-03-26 15:29:11.039707485 +0000 UTC (took 15.5s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:40447
sql:        postgresql://root@127.0.0.1:33683?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:     restarted pre-existing node
clusterID:  7eab080d-d52d-4336-baae-332a937d264f
nodeID:     2
CockroachDB node starting at 2018-03-26 15:29:22.547685106 +0000 UTC (took 4.7s)
build:               CCL v2.0-alpha.20180212-1888-g4b166b7 @ 2018/03/26 15:15:43 (go1.10)
admin:               http://127.0.0.1:46169
sql:                 postgresql://root@127.0.0.1:37841?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1/cockroach-temp942803217
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/1
status:              restarted pre-existing node
clusterID:           7eab080d-d52d-4336-baae-332a937d264f
nodeID:              1
CockroachDB node starting at 2018-03-26 15:29:24.299732675 +0000 UTC (took 0.7s)
build:               CCL v2.0-alpha.20180212-1888-g4b166b7 @ 2018/03/26 15:15:43 (go1.10)
admin:               http://127.0.0.1:45079
sql:                 postgresql://root@127.0.0.1:35899?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2/cockroach-temp516241572
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/2
status:              restarted pre-existing node
clusterID:           7eab080d-d52d-4336-baae-332a937d264f
nodeID:              3
CockroachDB node starting at 2018-03-26 15:29:26.334185802 +0000 UTC (took 0.6s)
build:               CCL v2.0-alpha.20180212-1888-g4b166b7 @ 2018/03/26 15:15:43 (go1.10)
admin:               http://127.0.0.1:41959
sql:                 postgresql://root@127.0.0.1:33935?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3/cockroach-temp821448691
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster056309001/3
status:              restarted pre-existing node
clusterID:           7eab080d-d52d-4336-baae-332a937d264f
nodeID:              2
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade109470162

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/show_cluster_setting.go
+++ b/pkg/sql/show_cluster_setting.go
@@ -62,7 +62,7 @@ func (p *planner) showStateMachineSetting(
 		// value, and the corresponding sql migration that makes sure
 		// the above select finds something usually runs pretty quickly
 		// when the cluster is bootstrapped.
-		kvRawVal, obj, err := s.Validate(&st.SV, prevRawVal, nil)
+		kvRawVal, kvObj, err := s.Validate(&st.SV, prevRawVal, nil)
 		if err != nil {
 			return errors.Errorf("unable to read existing value: %s", err)
 		}
@@ -70,11 +70,16 @@ func (p *planner) showStateMachineSetting(
 		// NB: if there is no persisted cluster version yet, this will match
 		// kvRawVal (which is taken from `st.SV` in this case too).
 		gossipRawVal := []byte(s.Get(&st.SV))
+
+		_, gossipObj, err := s.Validate(&st.SV, gossipRawVal, nil)
+		if err != nil {
+			gossipObj = fmt.Sprintf("<error: %s>", err)
+		}
 		if !bytes.Equal(gossipRawVal, kvRawVal) {
-			return errors.Errorf("gossip and KV store disagree about value")
+			return errors.Errorf("value differs between gossip (%v) and KV (%v); try again later", gossipObj, kvObj)
 		}
 
-		d = tree.NewDString(obj.(fmt.Stringer).String())
+		d = tree.NewDString(kvObj.(fmt.Stringer).String())
 		return nil
 	}); err != nil {
 		return nil, err

